ts/core/Tree/Node.ts

```diff
@@ -264,6 +269,18 @@ export abstract class AbstractNode implements Node {
   }
 
 
+  /**
+   * @override
+   */
+  public copy() {
+    const node = (this as AbstractNode).factory.create(this.kind) as AbstractNode;
+    node.properties = {...this.properties};
+    for (const child of this.childNodes || []) {
+      if (child) node.appendChild(child.copy());
```

Comment:
Children can be empty in case of incomplete `subsup`, `underover`, etc. 
Maybe add a comment.

Reply:
I have added a separate commit to address this.  The copied nodes were not being added to the various node lists in the `ParseOptions` object, and so the `msubsup` was not being converted to `msub` or `msup`.  So I've made a `ParseUtil` method `copyNode()` that calls `copy()` and then adds the new nodes to the proper lists.  It also modified `addNode()` to record if the node is added to a list that is other than the one for its node type (e.g., the `nonscript` list) so that its copies can be added to the same lists.